### PR TITLE
Add quorum config to infra fetchProvider

### LINKED ISSUE #868: infra's fetchProvider should return a FallbackProvider
https://github.com/abacus-network/abacus-monorepo/blob/75b116cd564a4112741d09c8b961f6d8f8931bd1/typescript/infra/src/config/chain.ts#L13

The secret will contain a comma separated list of provider rpc urls

### PR DESCRIPTION
**Pull request by yorhodes**
Fixes #868

- Requires provider URLs as JSON array with plural gcp secret name (eg `mainnet-rpc-endpoints-ethereum`)

### CODE REVIEW

**Review by tkporter — APPROVED** [review]

**Comment by nambrot** [comment]
I had assumed we would just use the existing secret and comma separate values in it?

**Comment by tkporter** [comment]
I'm kinda against doing that, I think creating a new secret is the best way. Using the same secret would require us to move over all consumers of the secret at the same time and wouldn't allow us to roll back to a prior version that can't handle commas if we ever need to

**Comment by nambrot** [comment]
Which consumers do we have? I don't really feel strongly, but what about the JSON formatting then? I.e. other tickets like https://github.com/abacus-network/abacus-monorepo/issues/869 assume comma separation right now

**Comment by tkporter** [comment]
I'm not too passionate about the formatting, whatever's easiest really. I think on the rust side I think we should pass in the values in whatever way serde expects for `Vec<String>`s, and if necessary then in Helm templates we can read the secrets either as JSON (you can parse json in the template) or comma separated values (you can split by commas in the template) and then construct the expected env vars from those via templates

File: typescript/infra/src/agents/index.ts [hunk 1]
```diff
@@ -262,8 +262,12 @@ export async function getSecretAwsCredentials<Chain extends ChainName>(
 export async function getSecretRpcEndpoint(
   environment: string,
   chainName: ChainName,
+  quorum = false,
```
[the remark this answers is not in the document]
Comment (yorhodes) [reply typescript/infra/src/agents/index.ts:265]: only without default params which is API breaking

**Comment by mattiecnvr** [comment]
@tkporter 
The expected way for serde-json to get `Vec<String>` would be `["thing1", "thing2"]`. A JSON array is the equiv. Counterpoint though is that it is backwards compatible if we just take a string and split by commas since the current one would then just be treated as a list of one element.

**Pull request merged**